H2MLauncher.UI/ViewModels/ServerBrowserViewModel.cs

```diff
@@ -439,14 +514,26 @@ await Task.Run(() => _gameServerCommunicationService.StartRetrievingGameServerIn
         }
     }
 
-    private void JoinServer(ServerViewModel? serverViewModel)
+    private void JoinServer(IServerViewModel? serverViewModel)
     {
         if (serverViewModel is null)
             return;
 
-        StatusText = _h2MCommunicationService.JoinServer(serverViewModel.Ip, serverViewModel.Port.ToString())
+        bool hasJoined = _h2MCommunicationService.JoinServer(serverViewModel.Ip, serverViewModel.Port.ToString());
+        
+        AddRecentToSettings(new RecentServerInfo()
+        {
+            Joined = DateTime.Now,
+            ServerIp = serverViewModel.Ip,
+            ServerName = serverViewModel.HostName,
+            ServerPort = serverViewModel.Port
+        });
+
+        StatusText = hasJoined
             ? $"Joined {serverViewModel.Ip}:{serverViewModel.Port}"
             : "Ready";
+
+        RefreshServersCommand.Execute(this);
```

Comment:
Question: why refresh after join?

Reply:
Well, main reason was, that in the case the user didn't join the server. The list was at least more up to date. And that the recents is updated, as it isn't added there before it is refreshed. Because it is added in LoadServersAsync I think.

Reply:
I see. We can fix that by just toggling a refresh of the property. I think the issue was also with the implicit conversion creating two instances idk